.github/workflows/tests.yml

```diff
@@ -310,6 +310,26 @@ jobs:
     needs: linting-done
     runs-on: ubuntu-latest
 
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # GHA requires all matrix configurations to have at least one value. We don't want to set one here though.
+          - _: monolith
+
+          # Test with workers
+          - workers: workers
+            regex: 'A-Ea-e'
+
+          - workers: workers
+            regex: 'F-Lf-l'
+
+          - workers: workers
+            regex: 'M-Qm-q'
+
+          - workers: workers
+            regex: 'R-Zr-z'
```

Comment:
I have reservations over this is a useful thing to do.

We are constrained as an organisation as to the number of GHA jobs we can run concurrently (to 60, according to https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#usage-limits). Splitting the tests up this way does nothing to reduce the total amount of GHA worker time used by our tests, but does increase the chances of us monopolising the workers.

An alternative might be to exclude worker-mode-complement from the `tests-done` job, so we don't need to wait for it. But that's fiddly too.

le sigh.

Reply:
Right, fair enough, I didn't realise we had this limit (well, I assumed there must be a limit, but I didn't know how much it was or how concerned we were about it).
You're right that this only reduces the wall time, at the cost of increasing the total consumed time (since each image has the overheads of building the Complement images, etc.).

Perhaps it may make sense, for now, to:
- restrict these worker tests to running on `develop` only(?) — gives us an indication if nothing else. Also prevents us having to 'wait' for it.
- try and get some speed improvements landed (e.g. the forking launcher, which cuts it down to ~26 minutes which is only a few minutes longer than the next slowest thing.)

Whilst testing, I've been cheating a bit and making Complement start before the lints are done; this gives it a few minutes headstart and then it's not the slowest thing on the list. If we're interested that much in shaving some time off, it could be worth considering, though I'd rather get some real time shavings if possible.

Reply:
> Perhaps it may make sense, for now, to:

yeah, let's try what you suggest for now

Reply:
Turns out you can't use a `matrix` in the `if` block of a job (YAML anchors are no good either), so I've duplicated it for now :/

Reply:
> Turns out you can't use a `matrix` in the `if` block of a job (YAML anchors are no good either), so I've duplicated it for now :/

Are you sure about this? We have examples of doing so, e.g. 

https://github.com/matrix-org/synapse/blob/57fac2a2347e0b386385868cdd1ae435364062b5/.github/workflows/latest_deps.yml#L58

Reply:
Oh, perhaps you mean the job-level if rather than the step-level if